src/utils/oauth2_util.py

```diff
@@ -473,3 +479,30 @@ def build_authentication_response(secret_key, access_token, refresh_token):
         authentication_response.update({"user": decoded_token.get("user")})
 
     return authentication_response
+
+
+def build_refresh_authentication_response(secret_key, access_token):
+    decoded_token = jwt.decode(access_token, secret_key, algorithms="HS256")
+
+    authentication_response = {
+        "access_token": access_token,
+        "expires_in": get_time_in_seconds_from_unix_time(decoded_token.get("exp")),
+        "client_name": decoded_token.get("client_name"),
+        "scopes": decoded_token.get("scopes"),
+    }
+
+    return authentication_response
+
+
+def get_request_token(request):
```

Comment:
This logic is wrong, the logic shoould be:
```
if request.url_rule == 'refresh_token_endpoint' and request.method == 'POST':
    refresh_token_request = request.get_json()
    return refresh_token_request["refresh_token"

authorization_header = request.headers.get("Authorization")
authorization_header_split = authorization_header.split(" ")
access_token = authorization_header_split[1]
return access_token
```

Reply:
Done.